js/packages/teams-ai/src/Application.ts

```diff
@@ -61,6 +63,11 @@ export interface ApplicationOptions<TState extends TurnState> {
      */
     adapter?: BotAdapter;
 
+    /**
+     * Optional. OAuth prompt settings to use for authentication.
+     */
+    authentication?: OAuthPromptSettings;
```

Comment:
I’m not sure whether using “OAuthPromptSettings” directly as authentication config will confuse developers, especially when they have some knowledge about BotFramework. For example, they might feel it's for bot authentication only. On the opposite, for a developer that only want to build a Message Extension and don't know BotFramework, the extra settings might also be confusing because they don't work (I feel only `ConnectionName` is valid to Message Extension).

Do you think reorganizing the config object is better? Here's a rough thought in my mind to start the brainstorm:
``` json
{
  "connectionName": "xxx",
  "botAuthenticationSettings": {
    "title": "..."
  },
  "messageExtensionAuthenticationSettings": {
    // Should be no settings for ME, we can remove this field. Put it here to better share my thought
  }
}
```

Reply:
I have made some major refactors since this comment. Let me know what you think about the current shape of `AuthenticationOptions`.